compiler/damlc/daml-lf-conversion/src/DA/Daml/LFConversion.hs

```diff
@@ -544,8 +545,27 @@ convertClassDef env tycon
             EBuiltin (BEText "undefined") -- We only care about the type, not the expr.
         funDepDef = defValue tycon (funDepName tsynName, funDepType) funDepExpr
 
-    pure $ [typeDef] ++ [funDepDef | classHasFds cls && newStyle]
-        -- NOTE (SF): No reason to generate fundep metadata with old-style typeclasses,
+    let minimal = fmap getOccText (classMinimalDef cls)
+        methodsWithNoDefault = sort [ getOccText id | (id, Nothing) <- classOpItems cls ]
+            -- ^ Used when MINIMAL pragma is not given,
+            -- i.e. the minimal sig is all methods without a default implementation.
+        minimalIsDefault =
+            case minimal of
+                BF.Var x -> [x] == methodsWithNoDefault
+                BF.And subclauses
+                    | names <- [ name | BF.Var name <- map unLoc subclauses ]
+                    , length names == length subclauses
+                    -> sort names == methodsWithNoDefault
```

Comment:
Why do we need all this special casing for default handling? Isn’t not specifying a minimal pragma equivalent to all methods being part of the default pragma and we can just aways reconstruct the pragma?

Reply:
nvm, of course it isn’t equivalent.

Reply:
A good reason is to avoid encoding the minimal pragmas unless we have to, to keep DAR size down. A certain codebase has a lot of typeclasses...